pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -102,10 +101,7 @@ func NewController(spec *SubmarinerIpamControllerSpecification, config *Informer
 	return ipamController, nil
 }
 
-func (i *Controller) Run(stopCh <-chan struct{}) error {
-	defer utilruntime.HandleCrash()
-	defer cleanup.ClearGlobalnetChains()
```

Comment:
When the active gateway migrates to a new node, the ipamController will be stopped and we want the cleanup to happen. Can you please point me where the cleanup is handled in this change?

Reply:
`cleanup.ClearGlobalnetChains` is now called by `GatewayMonitor.stopIpamController` and the chain cleanup is verified by the first test spec on L59 in _gatewaymonitor_test.go_. I made this function synchronous b/c it really didn't need to be async and keep a goroutine blocked until stopped, but, more importantly, it was problematic for the unit tests, ie it caused intermittent failures due to an issue in the fake clients (they're a bit flaky with concurrency on startup).

Reply:
Cool, thanks @tpantelis